### PR TITLE
Bring atomic_switch_spec code coverage to 100%

### PR DESCRIPTION
**Pull request by bbuchalter**
I am considering refactoring the atomic_switch's
retry behavior. Before doing so, let's make sure we
have great test coverage everywhere. This adds
some tools to make that happen and adds test
coverage.

### CODE REVIEW

**Comment by bbuchalter** [comment]
By luck, this branch discovered a flakey test which can be seen by running `TESTOPTS="--seed=57915" bundle exec rake integration`. I've opened an issue https://github.com/Shopify/lhm/issues/35. Before merging this branch, I'd like to address the flake, as it may not be easy to reproduce once the branch leaves this state, and this change doesn't block other planned work immediately.

File: spec/integration/atomic_switcher_spec.rb [hunk 1]
```diff
@@ -27,17 +28,22 @@
       Thread.abort_on_exception = false
     end
 
-    it 'should retry on lock wait timeouts' do
+    it 'should retry and log on lock wait timeouts' do
       skip 'This spec only works with mysql2' unless defined? Mysql2
+      connection = mock()
```
Comment (insom) [inline spec/integration/atomic_switcher_spec.rb]: Is this code accidentally duplicated?

**Comment by bbuchalter** [comment]
@insom great catch on the duplication. Was caused by a rebase. It's removed.

Sorry about introducing simplecov in the same commit as the test change. You are 100% correct that those should have been separate.

I've added one more commit to this PR which further simplifies `AtomicSwitcher` and now together with https://github.com/Shopify/lhm/pull/37 means that all three places where we retry now are a simple `connection.execute` statement.

File: lib/lhm/atomic_switcher.rb [hunk 2]
```diff
@@ -51,13 +45,11 @@ def validate
 
     def execute
       begin
-        statements.each do |stmt|
```
Comment (insom) [inline lib/lhm/atomic_switcher.rb]: Just smiling at the amount of pointless indirection you've removed...

**Comment by bbuchalter** [comment]
Rebased to address merge conflict.

**Pull request merged**